corehq/ex-submodules/couchforms/models.py

```diff
@@ -96,9 +96,11 @@ class XFormOperation(DocumentSchema):
     operation = StringProperty()  # e.g. "archived", "unarchived"
 
 
-class XFormInstance(SafeSaveDocument, UnicodeMixIn, ComputedDocumentMixin,
-                    CouchDocLockableMixIn, AbstractXFormInstance):
+class XFormInstance(DeferredBlobMixin, SafeSaveDocument, UnicodeMixIn,
```

Comment:
The requirement for this comes from the form processing where we want the processing to be atomic for each form submission. If we can approach that problem differently then I think we can get rid of the need for deferred attachments.

The only reason to use `Class.get_db().bulk_save(docs)` is if you're saving docs that have a different `doc_type` from the `Class` (or in generic doc processing code e.g. `IterDB`) . In our code this mostly happens when we do a soft delete (where we append `-deleted` to the `doc_type`). The other place it happens is for `XFormInstance` objects which can have a range of `doc_types`: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/couchforms/models.py#L50-L50